### PR TITLE
translate traits

### PR DESCRIPTION
**Pull request by yodalee**
translate trait

### CODE REVIEW

File: src/traits.md [hunk 1]
```diff
@@ -404,14 +384,13 @@ fn inverse<T>() -> T
 }
 ```
 
-This shows off the additional feature of `where` clauses: they allow bounds
-on the left-hand side not only of type parameters `T`, but also of types (`i32` in this case). In this example, `i32` must implement
-`ConvertTo<T>`. Rather than defining what `i32` is (since that's obvious), the
-`where` clause here constrains `T`.
+這展示 `where` 額外的特性，它們除了型別參數 `T`，亦可限制型別（範例中的 `i32`)
+在範例中，`i32` 必須實作 `ConvertTo<T>`，這裡的 `where` 限制 `T`而不是定義 `i32`是什麼
```
Comment (yodalee) [inline src/traits.md:388]: This section, I'm not sure how to translate it.

File: src/traits.md [hunk 2]
```diff
@@ -1,10 +1,9 @@
-% Traits
+# Traits
 
-A trait is a language feature that tells the Rust compiler about
-functionality a type must provide.
+trait 為Rust 語言特徵，告知Rust compiler 一個型別必須滿足的功能性
```
Comment (askeing) [inline src/traits.md]: 英文前後請留一個空白 `為 Rust 語言特徵`，句尾請加上全形句號 `。`
ref: [中文文案排版指南](https://github.com/sparanoid/chinese-copywriting-guidelines)

File: src/traits.md [hunk 3]
```diff
@@ -1,10 +1,9 @@
...
 
-Recall the `impl` keyword, used to call a function with [method
-syntax][methodsyntax]:
+記得提過的 `impl` 關鍵字，用在使用[method syntax][methodsyntax]
```
Comment (askeing) [inline src/traits.md]: link 的前後請加空白，另外句尾也需要句號。

File: src/traits.md [hunk 4]
```diff
@@ -43,15 +42,14 @@ impl HasArea for Circle {
...
+可以看到，`trait` 區塊和 `impl` 區塊看起來十分相似，但只需定義型別特徵，不需要定義方法本體
+當我們 `impl` 一個trait 時，使用 `impl Trait for Item` 而非 `impl Item`。
 
-## Trait bounds on generic functions
+## 泛型函式的 Trait 限制
 
-Traits are useful because they allow a type to make certain promises about its
-behavior. Generic functions can exploit this to constrain, or [bound][bounds], the types they
-accept. Consider this function, which does not compile:
+Traits 保證一個型別應有的行為，因此非常有用，泛型函式能利用 trait 作為[限制(bound)][bounds]
```
Comment (askeing) [inline src/traits.md]: 使用全形括號，以及加上全形逗號，建議可以改為這樣：

``` markdown
Traits 保證一個型別應有的行為，因此非常有用。
泛型函式能利用 trait 作為 [限制][bounds]（bound），用來限制它接受的型別。
```

File: src/traits.md [hunk 5]
```diff
@@ -253,15 +243,15 @@ let result = f.write(buf);
 # result.unwrap(); // ignore the error
 ```
 
-Here’s the error:
+以下為編譯錯諤：
```
Comment (askeing) [inline src/traits.md]: typo here

File: src/traits.md (hunk 1 above)
Comment (askeing) [inline src/traits.md:388]: I am also confused at `Rather than defining what i32 is (since that's obvious), the where clause here constrains T.`

Comment (yodalee) [inline src/traits.md:388]: 我覺得我先保留原文好了

**Comment by yodalee** [comment]
mostly fixed

**Comment by askeing** [comment]
@yodalee thx :)

**Pull request merged**